packages/manager/apps/pci-block-storage/src/api/hooks/useVolume.tsx

```diff
@@ -202,24 +202,31 @@ export const useAttachVolume = ({
   };
 };
 
+interface DetachParams {
+  projectId: string;
+  volumeId: string;
+  instanceId: string;
+}
+
 export const useDetachVolume = ({
-  projectId,
-  volumeId,
-  instanceId,
-  onError,
   onSuccess,
-}: AttachVolumeProps) => {
+  onError,
+}: {
+  onSuccess: () => void;
+  onError: (error: Error) => void;
+}) => {
   const mutation = useMutation({
-    mutationFn: async () => detachVolume(projectId, volumeId, instanceId),
+    mutationFn: async ({ projectId, volumeId, instanceId }: DetachParams) =>
+      detachVolume(projectId, volumeId, instanceId),
     onError,
-    onSuccess: (volume: TVolume) => {
+    onSuccess: (volume: TVolume, { projectId, instanceId, volumeId }) => {
```

Comment:
Will be better to abstract the logic from the mutation success handler, to enhance testability and scalability.

Reply:
I don't understand what you want to abstract here, these are the parameters required to do a mutation

Reply:
I was talking about the logic in the function, sorry for the confusion: 

```ts
queryClient.setQueryData(
         getVolumesQueryKey(projectId),
         (data: { id: string; attachedTo: string }[] | undefined) =>
           data?.map((v) => {
             if (v.id === volumeId) {
               return newVolume;
...
```